app/assets/javascripts/revised/pages/bikes/edit.coffee

```diff
@@ -17,6 +18,3 @@ class BikeIndex.BikesEdit extends BikeIndex
 
   submitBikeEditForm: ->
     $('form.primary-edit-bike-form').submit()
-    setTimeout (-> # Sometimes the page reload isn't triggered. Do it manually
-      location.reload(true)
-    ), 300
```

Comment:
This interferes with server-determined redirects -- I'm testing to see if it's safe to omit.

Reply:
Hmmmm. which server determined redirects?

Reply:
On master:

```rb
# app/controllers/bikes_controller.rb L172-177 (721391b4)

  else
    flash[:success] = "Bike successfully updated!"
    return if return_to_if_present
    redirect_to edit_bike_url(@bike, page: params[:edit_template]) and return
  end
end
```


This branch:
```rb
# app/controllers/bikes_controller.rb L196-202 (ab3fa27b)

  # When reporting a bike as stolen, redirect to the recovery form.
  # The recovery form redirects to the stolen form and is handled by
  # CoffeeScript in pages/bikes/edit_stolen.coffee
  target_template = (params[:edit_template] == "report_stolen") ?
    "report_recovered" : params[:edit_template]
  redirect_to edit_bike_url(@bike, page: target_template) and return
end
```

Reply:
Try commenting it out with a note and we can test this once it's live and see if it's necessary?

Reply:
It's necessary on this branch, to flip between the stolen and recovery forms and vice versa.